PiwikTracker/Dispatcher.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+internal protocol Dispatcher {
+    associatedtype Dispatchable
+    
+    var userAgent: String? { get set }
+    
+    func send(event: Dispatchable, success: ()->(), failure: (_ shouldContinue: Bool)->());
```

Comment:
I think constraining the Dispatchable to a protocol that serves a URLRequest is not really what we want. In the end the Event should not define its own url. Especially if we keep in mind the second method that dispatches multiple events at once via just one URLRequest. I will refactor the Dispatchable to use Event directly.